src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -4192,10 +4195,14 @@ def add_functionapp_vnet_integration(cmd, name, resource_group_name, vnet, subne
     functionapp = get_functionapp(cmd, resource_group_name, name)
     parsed_plan_id = parse_resource_id(functionapp.server_farm_id)
     plan_info = client.app_service_plans.get(parsed_plan_id['resource_group'], parsed_plan_id['name'])
+    app = _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'get', slot, client=client)
     if plan_info is None:
         raise ResourceNotFoundError('Could not determine the current plan of the functionapp')
     if is_plan_consumption(cmd, plan_info):
         raise ValidationError('Virtual network integration is not allowed for consumption plans')
+    if app.managed_environment is not None:
+        raise UnsupportOperationError('Config vnet for a function app on container app environment is not supported', 
+        'Please config your vnet when create the container app environment')
```

Comment:
As mentioned in a previous comment, you could make this a validator function and have the `validator` value in commands.py